Bug 1797789: [Doc] Update doc for adding "fieldDependency" specDescriptor (#4114)
openshift/console: frontend/packages/operator-lifecycle-manager/src/components/descriptors/reference/reference.md

</table>

### 20. Field Dependency

**x-descriptors**

> This descriptor allows you to specify a field as the dependent of a Control Field and being visible/present only when the current value of the Control Field is equal to the expected value:

Reviewer: ```suggestion
This descriptor allows you to specify a field as the dependent of a Control Field and shows the field when the current value of the Control Field is equal to the expected value:
```

```

**Usage**
1. Add “fieldDependency” to 'x-descriptors' array of the Dependent Field(s).
2. Replace “CONTROL_FIELD_PATH” with the 'path' property of the Control Field object.
3. Replace “EXPECTED_VALUE” with the actual expected value.

The Dependent Field(s) will be displayed only when the current value of the Control Field is equal to the expected value.

**Example**
```yaml
…
- displayName: Enable Upgrades
  description:  >-
      Set true to enable automatic micro version product upgrades, it is disabled by default.
  path: upgrades.enabled
  x-descriptors:
      - 'urn:alm:descriptor:com.tectonic.ui:booleanSwitch'
- displayName: Include minor version upgrades
  description: >-
      Set true to enable automatic minor product version upgrades, it is
      disabled by default. Requires spec.upgrades.enabled to be true.
  path: upgrades.minor
  x-descriptors:
      - 'urn:alm:descriptor:com.tectonic.ui:fieldDependency:upgrades.enabled:true'
      - 'urn:alm:descriptor:com.tectonic.ui:booleanSwitch'
…
```

**UI**
<table style="width:100%">
  <tr style="vertical-align:top">
    <td width="50%">CREATION VIEW
      <img src="img/20-1_fielddependency-new.png" />
      <br>

>       <small><b>* </b><i> When "Enable Upgrade" (CONTROL FIELD) is TRUE:</i></small>

Reviewer: ```suggestion
      <small><b>* </b><i> When "Enable Upgrades" (CONTROL FIELD) is TRUE:</i></small>
```
